### PR TITLE
Non-record: LapushBaby stock baseline 1xGPU RunPod

### PR DESCRIPTION
**Pull request by LapushBaby**
## Summary

This is a non-record submission documenting a reproducible single-GPU RunPod run using the stock root `train_gpt.py` copied into the submission folder.

Key result:
- `final_int8_zlib_roundtrip_exact val_bpb: 1.36294332`
- `val_loss: 2.30127271`
- total submission size: `12,264,252` bytes

## Setup

- Dataset: FineWeb `sp1024`
- Train shards: 10
- Hardware: 1x GPU on RunPod
- Wallclock cap: 600 seconds

## Repro command

```bash
RUN_ID=real_10min \
DATA_PATH=./data/datasets/fineweb10B_sp1024/ \
TOKENIZER_PATH=./data/tokenizers/fineweb_1024_bpe.model \
VOCAB_SIZE=1024 \
MAX_WALLCLOCK_SECONDS=600 \
VAL_LOSS_EVERY=0 \
torchrun --standalone --nproc_per_node=1 train_gpt.py

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Non-record: LapushBaby stock baseline 1xGPU RunPod

**Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache


## PR #630 — LapushBaby StockBaseline 10min 1GPU RunPod

**Head SHA:** 071d2200d176cd7f5db02779a2d2a4dac27c8e04  
**File audited:** `records/track_non_record_16mb/2026-03-24_LapushBaby_StockBaseline_10min_1GPU_RunPod/train_gpt.py`  
**Blob SHA:** 651beb2b895c3ff8ba4a232914d8b3772baaa868 (identical to repo root `train_gpt.py`)

---

### Check 1: N-gram family bug (CLOSE trigger)

No n-gram, bigram, trigram, hash table, or token lookup-key structures present anywhere in the file. The model is a standard GPT with `nn.Embedding`, `CausalSelfAttention`, and `MLP` blocks. No hash-based n-gram component exists. **CLEAN.**

### Check 2: Pre-Quant TTT (CLOSE trigger)

No test-time training of any kind present. There is no AdamW or any optimizer applied to `val_tokens`. Validation (`eval_val`) runs in `model.eval()` mode with `loss.detach()` — gradients are never computed on val data. No multi-epoch loop over val tokens anywhere. **CLEAN.**

### Check 3: Legal TTT (score-first-per-chunk)

No TTT at all — neither legal nor illegal. Not applicable.

### Check 4: Scored-region SLOT (HOLD trigger)

No scored-region / SLOT pattern detected. The submission is filed under `track_non_record_16mb` explicitly labeling itself a non-record stock baseline run on 1 GPU (RunPod), not the 8×H100 competition track. No SLOT trigger. **CLEAN.**

### Check 5: Pure neural

This is a clean pure-neural submission. Architecture: 9-layer GPT, vocab 1024, seq 1024, GQA (8 heads / 4 KV), relu² MLP, RoPE, Muon optimizer for matrix params, Adam for embeddings/scalars. No non-neural auxiliary structures. **CLEAN.**

---

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). Compliance audit via LLM agent (Sonnet) reviewing full train_gpt.py source. If this review misread your code, please call it out so I can re-audit manually.*
